docs/2-Interviews/answering_technical_questions.md

```diff
@@ -1,39 +1,129 @@
-- explain your thoughts
-  - just keep talking, even if you're not sure if you're right. they want to know how you struggle just as much as (if not more than) they want to see you just "get it"
-  - before you start coding, explain approach/flow in mind and ask if you’re on the right track/same page
-  - write stuff down as you think so they can see you can organize your thoughts
-- get a good idea of the design of how it’s going to work before you code it, especially if the coding will be easy but coming up with the right code will be hard (which is an ideal question)
-- keep a good pace (keep in mind how long you have and how much you're aiming to get done ... but not too much cause it can be sort of terrifying)
-- it’s okay to start with brute force! just get started, you can optimize things later - better to get a simple solution than no solution
-- it’s definitely not about getting the right answer right away - any good coder will eventually write code to do it - it’s about getting there within time restraints and the process that got you there
-- once you get a draft going, ask if any further optimizations are necessary
-  - are you optimizing for time? space? which is more important?
-- write/talk through some test cases if you have time, to prove to yourself and the interviewer that your code works
-  - consider [edge cases](http://softwareengineering.stackexchange.com/questions/72761/how-do-you-identify-edge-cases-on-algorithms)!
-
-Questions about your experience/projects
+### Foreword
+
+This section contains a lot of information and can look very daunting. **You don't need to know everything in here to interview!**. The good news is that a lot of these will probably come to you naturally as you practice and get experience interviewing. It's totally expected that the first few interviews are going to involve some learning! There's no substitute to practice. The goal of this section is **not to memorize it**, but to use it as a **guide to help you reflect** on your experience and help you improve in the long run.
+
+### Which language should I use?
+
+The best answer to this question is "whatever you're the most comfortable with". In 99% of algorithm interviews, they want to see if you understand programming concepts and can write code. You can show that to them in just about any language. Using the language you are most familiar with will minimize friction, such as unfamiliarity with the language and its libraries.
+
+That being said, some languages allow you to express concepts much more quickly and ergonomically than others, and eventually you'll want to get comfortable with one of those languages. We recommend Python in particular *for interviews* (though it's also a useful language to know in general).
+
+Things that can get in the way in other languages include:
+- Writing types (we love types, but you only have one hour to write your code and explain it!)
+- Manual memory management
+- Verbose syntax
```

Comment:
Verbose syntax (using Python will usually result in much less writing, which saves so much time)